components/Container/LayoutWithPanel.tsx

```diff
@@ -0,0 +1,237 @@
+'use client';
+
+import React, {
+  CSSProperties,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+
+import { Button } from 'react-aria-components';
+
+import { DmpIcon } from '@/components/Icons';
+import {
+  useResponsive,
+  getSizeByName,
+} from '@/hooks/responsive';
+
+import { LayoutContainerProps } from '@/components/Container/LayoutContainer';
+import { ContentContainer } from '@/components/Container/ContentContainer';
+
+
+type DirectionType =
+  | null
+  | "left"
+  | "right";
+
+
+/**
+ * Layout with a dynamic sidebar
+ */
+interface LayoutWithPanelProps extends LayoutContainerProps {
+}
+
+export const LayoutWithPanel: React.FC<LayoutWithPanelProps> = ({
+  children,
+  id,
+  className = "",
+}) => {
+  const thisRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (thisRef) {
+      if (thisRef.current) {
+        let direction: DirectionType = null;
+
+        function _updateClassList(name: string) {
```

Comment:
L46: @andrewebdev I see in error here because a "function declaration" is not allowed here, but you could convert this to a "function expression" like `const _updateClassList = (name: string) => {`